official/vision/beta/projects/panoptic_maskrcnn/modeling/factory.py

```diff
@@ -0,0 +1,244 @@
+  if roi_sampler_config.cascade_iou_thresholds:
+    for iou in roi_sampler_config.cascade_iou_thresholds:
+      roi_sampler_obj = roi_sampler.ROISampler(
+          mix_gt_boxes=False,
+          num_sampled_rois=roi_sampler_config.num_sampled_rois,
+          foreground_iou_threshold=iou,
+          background_iou_high_threshold=iou,
+          background_iou_low_threshold=0.0,
+          skip_subsampling=True)
+      roi_sampler_cascade.append(roi_sampler_obj)
+
+  roi_aligner_obj = roi_aligner.MultilevelROIAligner(
+      crop_size=roi_aligner_config.crop_size,
+      sample_offset=roi_aligner_config.sample_offset)
+
+  detection_generator_obj = detection_generator.DetectionGenerator(
+      apply_nms=generator_config.apply_nms,
+      pre_nms_top_k=generator_config.pre_nms_top_k,
+      pre_nms_score_threshold=generator_config.pre_nms_score_threshold,
+      nms_iou_threshold=generator_config.nms_iou_threshold,
+      max_num_detections=generator_config.max_num_detections,
+      use_batched_nms=generator_config.use_batched_nms)
+
+  mask_head = instance_heads.MaskHead(
+      num_classes=model_config.num_classes,
+      upsample_factor=model_config.mask_head.upsample_factor,
+      num_convs=model_config.mask_head.num_convs,
+      num_filters=model_config.mask_head.num_filters,
+      use_separable_conv=model_config.mask_head.use_separable_conv,
+      activation=model_config.norm_activation.activation,
+      norm_momentum=model_config.norm_activation.norm_momentum,
+      norm_epsilon=model_config.norm_activation.norm_epsilon,
+      kernel_regularizer=l2_regularizer,
+      class_agnostic=model_config.mask_head.class_agnostic)
+
+  mask_sampler_obj = mask_sampler.MaskSampler(
+      mask_target_size=(
+          model_config.mask_roi_aligner.crop_size *
+          model_config.mask_head.upsample_factor),
+      num_sampled_masks=model_config.mask_sampler.num_sampled_masks)
+
+  mask_roi_aligner_obj = roi_aligner.MultilevelROIAligner(
+      crop_size=model_config.mask_roi_aligner.crop_size,
+      sample_offset=model_config.mask_roi_aligner.sample_offset)
+
+  segmentation_head = segmentation_heads.SegmentationHead(
+      num_classes=model_config.num_classes,
+      level=segmentation_head_config.level,
+      num_convs=segmentation_head_config.num_convs,
+      prediction_kernel_size=segmentation_head_config.prediction_kernel_size,
+      num_filters=segmentation_head_config.num_filters,
+      upsample_factor=segmentation_head_config.upsample_factor,
+      feature_fusion=segmentation_head_config.feature_fusion,
+      low_level=segmentation_head_config.low_level,
+      low_level_num_filters=segmentation_head_config.low_level_num_filters,
+      activation=norm_activation_config.activation,
+      use_sync_bn=norm_activation_config.use_sync_bn,
+      norm_momentum=norm_activation_config.norm_momentum,
+      norm_epsilon=norm_activation_config.norm_epsilon,
+      kernel_regularizer=l2_regularizer)
+
```

Comment:
Maybe build maskrcnn model first using the maskrcnn factory funciton, then build the panotpic maskrcnn model by passing maskrcnn_model.backbone, maskrcnn_model.decoder, ... etc. to the panoptic_maskrcnn_model.PanopticMaskRCNNModel constructor below? What do you think?

Reply:
Yup, that would remove a lot of redundant code, I'll use the `build_maskrcnn` here